core/tests/protractor_desktop/preferences.js

```diff
@@ -26,6 +26,31 @@ describe('Preferences', function() {
     preferencesPage = new PreferencesPage.PreferencesPage();
   });
 
+  it('should upload a profile photo', function() {
+    users.createUser('eve@preferences.com', 'evePreferences');
+    users.login('eve@preferences.com');
+    preferencesPage.get();
+    expect(preferencesPage.getProfilePhotoSource())
+      .not
+      .toEqual(
+        preferencesPage.submitProfilePhoto('../data/img.png')
```

Comment:
@U8NWXD I think this will not work in prod_env, have you tried running the test with prod_env flag? (#6687 will be helpful, maybe?)

Reply:
@DubeySandeep this appears to work with the `prod_env` flag:

```console
$ bash scripts/run_e2e_tests.sh --prod_env --suite=preferences --sharding=false
...
Started
Jasmine started
.
  Preferences
    ✓ should upload a profile photo
.    ✓ shows an error if uploaded photo is too large
.    ✓ should change editor role email checkbox value
.    ✓ should change feedback message email checkbox value
.    ✓ should set and edit bio in user profile
.    ✓ should change prefered audio language of the learner
.    ✓ should change prefered site language of the learner
.    ✓ should load the correct dashboard according to selection




8 specs, 0 failures
...
```

Reply:
@U8NWXD, I tried running server locally in prod_env and tried uploading images, found that it fails with 500. Cna you [please check how it is passing e2e tests?

![image](https://user-images.githubusercontent.com/16653571/64486787-56adf080-d24f-11e9-85ab-2686973df558.png)

Reply:
Hi @U8NWXD, are you able to reproduce this issue? Let me know if you are facing any issue?

Reply:
@U8NWXD ping?

Reply:
Sorry for the delayed response; I've been off the grid for the past week or so. I'll work on reproducing and debugging this issue

Reply:
@DubeySandeep I was unable to reproduce this error. Here are the steps I followed:

* Run server locally with `python -m scripts.start`
* Select `Sign In` from the top menu bar
* Create an account with email `test@example.com` and username `test`
* Select `Create` from the top menu bar
* Select the introduction block for editing
* Choose to upload `oppia/core/tests/data/img.png`
* Set caption to `hi` and accessibility text to `foobar`
* Click `Use This Image`
* Click `Save`

No error occurred, and the image was added to the introduction block.

I was running after merging in `develop` such that the last non-merge commit is ec254cfcc87843c36e24d68d335cc811536fb2d4. I will update this PR with this merge. If you are still seeing this error on the updated branch, could you send me steps to reproduce?

Reply:
@U8NWXD, All the steps you have mentioned are correct except step 1, make it "Run server locally with `python -m scripts.start --prod_env`" as the e2e test on Travis runs in the prod mode.

Reply:
@DubeySandeep When running in prod mode on develop, I'm unable to create a user via the `Sign In` menu bar option. After choosing to use `test@example.com` as the email, I get a blank page at `http://localhost:8181/signup?return_url=%2F` instead of the page where I select a username and agree to the terms of service.

Based on the server output to the console, this appears to be because of a 404 when retrieving `third_party/generated/js/third_party.js`. The file that exists (and that I think should be the target) is `third_party/generated/js/third_party.min.js`.

Output:

```
...
INFO     2019-09-20 04:46:34,396 module.py:835] default: "GET /third_party/generated/css/third_party.css HTTP/1.1" 404 -
INFO     2019-09-20 04:46:34,465 module.py:835] default: "GET /templates/dev/head/css/oppia.css HTTP/1.1" 304 -
INFO     2019-09-20 04:46:34,606 module.py:835] default: "GET /webpack_bundles/signup.bundle.js HTTP/1.1" 304 -
INFO     2019-09-20 04:46:34,652 module.py:835] default: "GET /webpack_bundles/about%7Eadmin%7Eclassroom%7Ecollection_editor%7Ecollection_player%7Ecommunity_dashboard%7Econsole_errors%7Econtact%7Ee78591d4.bundle.js HTTP/1.1" 304 -
INFO     2019-09-20 04:46:34,746 module.py:835] default: "GET /webpack_bundles/runtime.bundle.js HTTP/1.1" 304 -
INFO     2019-09-20 04:46:35,703 module.py:835] default: "GET /webpack_bundles/vendors%7Eabout%7Eadmin%7Eclassroom%7Ecollection_editor%7Ecollection_player%7Ecommunity_dashboard%7Econsole_errors%7E561518cc.bundle.js HTTP/1.1" 304 -
INFO     2019-09-20 04:46:36,339 module.py:835] default: "GET /third_party/generated/js/third_party.js HTTP/1.1" 404 -
```

Reply:
@U8NWXD, Have you tried updating your branch with upstream/develop? (though I don't expect this to happen for an outdated branch as well.)

Reply:
@DubeySandeep Yes I have, but the problem persists, even after I ran
```
rm -rf ../oppia_tools/ node_modules/ third_party/ core/templates/prod/ local_compiled_js/
```
I'm now trying with a new clone of the oppia repo

Reply:
@DubeySandeep I was able to reproduce your error, and I filed #7686 for it. That error occurs when uploading a photo to an exploration, but not when uploading a profile photo. My e2e test passes despite the error because I'm only testing uploading profile photos, not uploading exploration photos.

Reply:
@DubeySandeep It seems to me that this bug is separate from (and not intended to be caught by) the e2e test introduced by this PR. Do you agree?

Reply:
> @DubeySandeep It seems to me that this bug is separate from (and not intended to be caught by) the e2e test introduced by this PR. Do you agree?

That's not a bug, it's expected behavior in prod_env we store images in GoogleCloudService but locally GCS key are not available so the upload fails.

I was not sure whether we upload profile images on GCS or in the app engine datastore itself. If we store images in AppEngine itself then I'm fine with this PR. (I'll check, make sure and approve this PR.)

Reply:
I confirm that we upload a base64 encoded profile image in the AppEngine itself. (This is only for the profile pictures, not for the other images such as images in exploration card.)